Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcCopyClient.cs

```diff
@@ -19,10 +21,40 @@ namespace BuildXL.Cache.ContentStore.Service.Grpc
     /// An implementation of a CAS copy helper client based on GRPC.
     /// TODO: Consolidate with GrpcClient to deduplicate code. (bug 1365340)
     /// </summary>
-    public class GrpcCopyClient : IShutdown<BoolResult>
+    public sealed class GrpcCopyClient : IShutdown<BoolResult>
     {
+        private static ConcurrentDictionary<(string, int), GrpcCopyClient> _clientPool = new ConcurrentDictionary<(string, int), GrpcCopyClient>();

```

Comment:
I would suggest to use named tuple. It is hard to tell from the definition what's the key.